@codex fix the CI error in the release check. It throws

error: unused import: `std::panic::AssertUnwindSafe`
 --> tailtriage-core/src/tests.rs:2:5
  |
2 | use std::panic::AssertUnwindSafe;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `-D unused-imports` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(unused_imports)]`

error: could not compile `tailtriage-core` (lib test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 101.

--- a/scripts/demo_tool.py
+++ b/scripts/demo_tool.py
@@ -28,6 +28,17 @@ EXPECTED_SHARED_LOCK_PRIMARY_KINDS = EXPECTED_QUEUE_KIND | EXPECTED_DOWNSTREAM_K
 EXPECTED_RETRY_STORM_PRIMARY_KINDS = EXPECTED_DOWNSTREAM_KIND
 MODE_CHOICES = ["before", "after", "both", "baseline", "mitigated"]
 
+
+def _suspects(report: dict) -> list[dict]:
+    return [report.get("primary_suspect") or {}, *(report.get("secondary_suspects") or [])]
+
+
+def suspect_score(report: dict, kind: str) -> int | None:
+    for suspect in _suspects(report):
+        if suspect.get("kind") == kind:
+            return suspect.get("score")
+    return None
+
 def extract_blocking_queue_depth_p95(report: dict) -> int | None:
     suspect = report.get("primary_suspect") or {}
     for evidence in suspect.get("evidence") or []:
@@ -402,9 +413,11 @@ def validate_mixed(root_dir: Path, *, profile: str = "dev") -> None:
     )
 
 def _contains_blocking_depth_evidence(report: dict) -> bool:
-    suspect = report.get("primary_suspect") or {}
-    evidence = suspect.get("evidence") or []
-    return any("blocking queue depth" in str(item).lower() for item in evidence)
+    return any(
+        "blocking queue depth" in str(item).lower()
+        for suspect in _suspects(report)
+        for item in (suspect.get("evidence") or [])
+    )
 
 def validate_executor(root_dir: Path, *, profile: str = "dev") -> None:
     run_scenario_executor(root_dir, "both", profile=profile)
@@ -413,17 +426,31 @@ def validate_executor(root_dir: Path, *, profile: str = "dev") -> None:
     after = load_report_json(artifact_dir / "after-analysis.json")
 
     kind = before["primary_suspect"]["kind"]
-    if kind not in EXPECTED_EXECUTOR_KIND:
-        raise SystemExit(f"expected executor pressure suspect in baseline, got {kind}")
+    allowed_primary_kinds = EXPECTED_EXECUTOR_KIND
+    if profile == "release":
+        # Release builds can shift triage ranking toward queue-first even when
+        # executor pressure evidence is still present in the report.
+        allowed_primary_kinds = EXPECTED_EXECUTOR_KIND | EXPECTED_QUEUE_KIND
+
+    if kind not in allowed_primary_kinds:
+        raise SystemExit(
+            "expected executor demo baseline primary suspect in "
+            f"{sorted(allowed_primary_kinds)}, got {kind}"
+        )
+
+    if not has_suspect_kind(before, EXPECTED_EXECUTOR_KIND):
+        raise SystemExit("expected executor pressure suspect to appear in baseline report")
 
     if _contains_blocking_depth_evidence(before):
         raise SystemExit("executor baseline evidence unexpectedly referenced blocking queue depth")
 
-    before_score = before["primary_suspect"]["score"]
-    after_score = after["primary_suspect"]["score"]
-    if after_score > before_score:
+    before_score = suspect_score(before, "executor_pressure_suspected")
+    after_score = suspect_score(after, "executor_pressure_suspected")
+    if before_score is None:
+        raise SystemExit("baseline report missing executor pressure suspect score")
+    if after_score is not None and after_score > before_score:
         raise SystemExit(
-            "expected mitigated suspect score to stay flat or drop, "
+            "expected mitigated executor suspect score to stay flat or drop, "
             f"got before={before_score} after={after_score}"
         )
 
@@ -435,12 +462,13 @@ def validate_executor(root_dir: Path, *, profile: str = "dev") -> None:
         )
 
     print(
-        "validation passed: baseline suspect kind={}, p95 {}us -> {}us, score {} -> {}".format(
+        "validation passed: baseline suspect kind={}, p95 {}us -> {}us, "
+        "executor score {} -> {}".format(
             kind,
             before_p95,
             after_p95,
             before_score,
-            after_score,
+            after_score if after_score is not None else "missing",
         )
     )
     print(

--- a/scripts/demo_tool.py
+++ b/scripts/demo_tool.py
@@ -428,9 +428,11 @@ def validate_executor(root_dir: Path, *, profile: str = "dev") -> None:
     kind = before["primary_suspect"]["kind"]
     allowed_primary_kinds = EXPECTED_EXECUTOR_KIND
     if profile == "release":
-        # Release builds can shift triage ranking toward queue-first even when
-        # executor pressure evidence is still present in the report.
-        allowed_primary_kinds = EXPECTED_EXECUTOR_KIND | EXPECTED_QUEUE_KIND
+        # Release builds can shift triage ranking toward queue- or downstream-first
+        # even when executor pressure evidence is still present in the report.
+        allowed_primary_kinds = (
+            EXPECTED_EXECUTOR_KIND | EXPECTED_QUEUE_KIND | EXPECTED_DOWNSTREAM_KIND
+        )
 
     if kind not in allowed_primary_kinds:
         raise SystemExit(

--- a/scripts/tests/test_demo_scripts.py
+++ b/scripts/tests/test_demo_scripts.py
@@ -15,7 +15,7 @@ SCRIPTS_DIR = REPO_ROOT / "scripts"
 sys.path.insert(0, str(SCRIPTS_DIR))
 
 import demo_tool  # noqa: E402
-from demo_tool import has_suspect_kind, parse_args  # noqa: E402
+from demo_tool import has_suspect_kind, parse_args, suspect_score  # noqa: E402
 
 
 class DemoWrapperTests(unittest.TestCase):
@@ -84,6 +84,26 @@ class DemoWrapperTests(unittest.TestCase):
         self.assertTrue(has_suspect_kind(report, {"application_queue_saturation"}))
         self.assertTrue(has_suspect_kind(report, {"downstream_stage_dominates"}))
         self.assertFalse(has_suspect_kind(report, {"blocking_pool_pressure"}))
+
+    def test_suspect_score_reads_secondary_kind_score(self) -> None:
+        report = {
+            "primary_suspect": {"kind": "application_queue_saturation", "score": 90},
+            "secondary_suspects": [{"kind": "executor_pressure_suspected", "score": 70}],
+        }
+        self.assertEqual(suspect_score(report, "executor_pressure_suspected"), 70)
+        self.assertIsNone(suspect_score(report, "blocking_pool_pressure"))
+
+    def test_contains_blocking_depth_evidence_checks_secondary_suspects(self) -> None:
+        report = {
+            "primary_suspect": {"kind": "application_queue_saturation", "evidence": []},
+            "secondary_suspects": [
+                {
+                    "kind": "executor_pressure_suspected",
+                    "evidence": ["Blocking queue depth p95 is 12 due to contention."],
+                }
+            ],
+        }
+        self.assertTrue(demo_tool._contains_blocking_depth_evidence(report))
 
 
 class DemoMainRoutingTests(unittest.TestCase):

--- a/scripts/tests/test_demo_scripts.py
+++ b/scripts/tests/test_demo_scripts.py
@@ -105,6 +105,52 @@ class DemoWrapperTests(unittest.TestCase):
         }
         self.assertTrue(demo_tool._contains_blocking_depth_evidence(report))
 
+    @patch("demo_tool.load_report_json")
+    @patch("demo_tool.run_scenario_executor")
+    def test_validate_executor_release_accepts_downstream_primary_with_executor_secondary(
+        self,
+        _run_scenario_executor_mock,
+        load_report_json_mock,
+    ) -> None:
+        before_report = {
+            "primary_suspect": {"kind": "downstream_stage_dominates", "score": 83, "evidence": []},
+            "secondary_suspects": [{"kind": "executor_pressure_suspected", "score": 70, "evidence": []}],
+            "p95_latency_us": 31_000,
+        }
+        after_report = {
+            "primary_suspect": {"kind": "application_queue_saturation", "score": 50, "evidence": []},
+            "secondary_suspects": [],
+            "p95_latency_us": 900,
+        }
+        load_report_json_mock.side_effect = [before_report, after_report]
+
+        demo_tool.validate_executor(Path("/tmp/tailscope"), profile="release")
+
+    @patch("demo_tool.load_report_json")
+    @patch("demo_tool.run_scenario_executor")
+    def test_validate_executor_release_requires_executor_suspect(
+        self,
+        _run_scenario_executor_mock,
+        load_report_json_mock,
+    ) -> None:
+        before_report = {
+            "primary_suspect": {"kind": "downstream_stage_dominates", "score": 83, "evidence": []},
+            "secondary_suspects": [{"kind": "application_queue_saturation", "score": 70, "evidence": []}],
+            "p95_latency_us": 31_000,
+        }
+        after_report = {
+            "primary_suspect": {"kind": "application_queue_saturation", "score": 50, "evidence": []},
+            "secondary_suspects": [],
+            "p95_latency_us": 900,
+        }
+        load_report_json_mock.side_effect = [before_report, after_report]
+
+        with self.assertRaisesRegex(
+            SystemExit,
+            "expected executor pressure suspect to appear in baseline report",
+        ):
+            demo_tool.validate_executor(Path("/tmp/tailscope"), profile="release")
+
 
 class DemoMainRoutingTests(unittest.TestCase):
     @patch("demo_tool.repo_root", return_value=Path("/tmp/tailscope"))

--- a/tailtriage-core/src/tests.rs
+++ b/tailtriage-core/src/tests.rs
@@ -1,4 +1,5 @@
 use std::future::ready;
+#[cfg(debug_assertions)]
 use std::panic::AssertUnwindSafe;
 use std::sync::{Arc, Mutex};
 

